### PR TITLE
bugfix: cover potential race condition in double create payment call

### PR DESCRIPTION
**Pull request by RKlompUU**
There was a race condition in the `/create-payment-intent` functionality.

Early in the execution path behind that, the API ensures that the user doesn't already have a payment intent active for the given payment provider, and if the user does it either:
- cancels the already active payment intent (if `recreateNftOrder` flag is true in the API call)
- throws an HttpError (if `recreateNftOrder` is false).

But this check wouldn't work if someone were to call the API in quick succession, where both executions get past this check before either of them commit their db transaction at the end: this is a race condition.

Postgres has a mode that prevents this kind of race condition (between reads and later writes): SERIALIZABLE mode. But it is not the default mode, and obviously makes queries run slower. 

Solution in this PR: set isolation level to SERIALIZABLE mode only for the create payment intent db transactions.

---

Note: 

Also noticed some errors with the recent Simplex integration in one of our deployments. the cronjob that checks for if Simplex has any event updates should only run if Simplex is enabled as a service, but it was trying to check this too if Simplex isn't enabled at all: resulted in errors because we're trying to call Simplex with undefined/default Simplex settings.

Solution: just return early from this cronjob function when there are 0 Simplex payment intents that are pending according to our own payment pg table.

### CODE REVIEW

**Review by ollyfreeman — COMMENTED** [review]
The two fixes look good to me.

Other than them, most of the diff on this PR was clean ups, all of which look good.

While I was looking at the clean-up parts of the diff, I noticed that there was quite a lot of inconsistencies (e.g. a) sometimes using template literals and sometimes using string concatenation, b) sometimes starting error messages with a capital letter, sometimes not), but those were pre-existing inconsistencies so I didn't comment on them.

But other than that, it looks good to me 👍

**Comment by umitduran** [comment]
these two fixes are necessary. You are right to prevent from race conditions. It make slows queries. But we have to do it. Thanks.

As @ollyfreeman mentioned that there are a lot of diff because of indent or function uses etc. Actually it depends on usages. When I make changes. My ide(webstorm) and settings make differently arranged. I guess We should find a common solution for these diffs. Correct me I am wrong that's why eslint file exists.

File: store-api-server/test/app.e2e.spec.ts [hunk 1]
```diff
@@ -1972,6 +1982,78 @@ describe('AppController (e2e)', () => {
     },
   );
 
+  skipOnPriorFail(
```
Comment (umitduran) [inline store-api-server/test/app.e2e.spec.ts:1985]: nice

**Comment by RKlompUU** [comment]
> these two fixes are necessary. You are right to prevent from race conditions. It make slows queries. But we have to do it. Thanks.
> 
> As @ollyfreeman mentioned that there are a lot of diff because of indent or function uses etc. Actually it depends on usages. When I make changes. My ide(webstorm) and settings make differently arranged. I guess We should find a common solution for these diffs. Correct me I am wrong that's why eslint file exists.

Just for visibility (we already got on the same page over Slack): it is why .prettierrc exists actually. eslint just provides warnings about probable code issues, prettier provides auto formatting.

**Pull request merged**